test/common/test_common.cpp

```diff
@@ -253,39 +274,298 @@ TEST (PCL, PointCloud)
+{
+  cloud.resize (640, 480);
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640);
+  EXPECT_EQ (cloud.height, 480);
+}
+
+TEST_F (pointCloudTest, resize_with_initialized_count_elements)
+{
+  cloud.resize (640*360, PointXYZ (1, 1, 1));
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640*360);
+}
+
+TEST_F (pointCloudTest, resize_with_initialized_count_and_new_width_and_height)
+{
+  cloud.resize (640, 480, PointXYZ (1, 1, 1));
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640);
+}
+
+TEST_F (pointCloudTest, assign_with_copies)
+{
+  cloud.assign (640*360, PointXYZ (1, 1, 1));
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640*360);
+}
+
+TEST_F (pointCloudTest, assign_with_new_width_and_height_copies)
+{
+  cloud.assign(640, 480, PointXYZ (1, 1, 1));
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640);
+}
+
+TEST_F (pointCloudTest, assign_with_copies_in_range)
+{
+  std::vector<PointXYZ> pointVec;
+  pointVec.resize (640*360, PointXYZ (2, 3, 4));
+  cloud.assign (pointVec.begin(), pointVec.end());
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640*360);
+}
+
+TEST_F (pointCloudTest, assign_with_copies_in_range_and_new_width)
+{
+  std::vector<PointXYZ> pointVec;
+  pointVec.resize (640*360, PointXYZ (2, 3, 4));
+  cloud.assign (pointVec.begin(), pointVec.end(), 640);
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640);
+}
+
+TEST_F (pointCloudTest, assign_mismatch_size_and_width_height)
+{
+  std::vector<PointXYZ> pointVec;
+  pointVec.resize (640*480, PointXYZ (7, 7, 7));
+  cloud.assign (pointVec.begin(), pointVec.end(), 460);
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 640*480);
+}
+
+TEST_F (pointCloudTest, assign_initializer_list)
+{
+  cloud.assign ({PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5)});
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 3);
+}
+
+TEST_F (pointCloudTest, assign_initializer_list_with_new_width)
+{
+  cloud.assign ({PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5)}, 2);
+  EXPECT_TRUE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 2);
+}
+
+TEST_F (pointCloudTest, assign_initializer_list_with_unorganized_cloud)
+{
+  cloud.assign ({PointXYZ (3, 4, 5), PointXYZ (3, 4, 5), PointXYZ (3, 4, 5)}, 6);
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 3);
+}
+
+TEST_F (pointCloudTest, push_back_to_unorganized_cloud)
+{
+  cloud.push_back (PointXYZ (3, 4, 5));
+  EXPECT_FALSE (cloud.isOrganized ());
+  EXPECT_EQ (cloud.width, 1);
+}
+
+TEST_F (pointCloudTest, push_back_to_organized_cloud)
+{
+  set_cloud(80*80, PointXYZ (1, 1, 1), 80);
+  EXPECT_TRUE (cloud.isOrganized ());
+  cloud.push_back (PointXYZ (3, 4, 5));
+  EXPECT_EQ (cloud.width, (80*80) + 1);
+}
+
+TEST_F (pointCloudTest, transient_push_back)
+{
+  set_cloud (640*480, PointXYZ (1, 1, 1), 640);
```

Comment:
Isn't this `cloud.resize (640, 480, PointXYZ (1, 1, 1));`?

If you split your test into 2 fixtures:
* ones before this
* ones with call to `set_cloud`

Then in the second one, you can put this in the setup (and test the resize function in the first fixture). 😄

Reply:
I am not sure an extra fixture will bring much to this test. I think it covers most of the methods and paths we're interested in.

Reply:
That's correct, but it'd remove the need to call a set_cloud function before every test case from this line onward

Reply:
That sounds reasonable. I can do that.